packages/ckeditor5-engine/src/view/placeholder.js

```diff
@@ -229,7 +229,10 @@ function updatePlaceholder( writer, element, config ) {
 		wasViewModified = true;
 	}
 
-	if ( needsPlaceholder( hostElement, config.keepOnFocus ) ) {
+	// If the host element is not a direct host then placeholder is needed only when there is only one element.
+	const isOnlyChild = isDirectHost || element.childCount == 1;
+
+	if ( isOnlyChild && needsPlaceholder( hostElement, config.keepOnFocus ) ) {
```

Comment:
Wouldn't the new check logic make more sense in `needsPlaceholder` method? It will require us to pass more variables but currently `needsPlaceholder` is the only place that determines whether placeholder is visible or not, with these changes we have two such places.

Reply:
I consider `needsPlaceholder` as a check for a single element (it's used in the Title plugin to check individual elements for the need for a placeholder). I would not propagate the `isDirectHost` option deeper because it's only complicating the code without adding any value.

Reply:
Good point on it being used in title plugin I assumed that it's only used in engine. OK, in this case it's good for now but as soon as we have more conditions like that, it make sense to push it to a separate function that also calls `needsPlaceholder`.